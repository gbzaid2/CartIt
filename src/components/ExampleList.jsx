import React from "react";
import { connect } from "react-redux";

const mapStateToProps = (state) => {
  return { data: state.data };
};

const ConnectedExampleList = (props) => {
  return (
    <ul>
      {props.data.map((item) => {
        return <li key={item.id}>{item.data}</li>;
      })}
    </ul>
  );
};

const ExampleList = connect(mapStateToProps)(ConnectedExampleList);

export default ExampleList;
