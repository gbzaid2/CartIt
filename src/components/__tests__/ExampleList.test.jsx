import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { createStore } from "redux";

import ExampleList from "../../components/ExampleList.jsx";
import rootReducer from "../../js/reducers/index";

import renderer from "react-test-renderer";

describe("ExampleList", () => {
  const mockStore = createStore(rootReducer, {
    data: [
      { id: 1, data: "Example Item #1" },
      { id: 2, data: "Example Item #2" }
    ]
  });

  const getWrapper = () =>
    mount(
      <Provider store={mockStore}>
        <ExampleList />
      </Provider>
    );

  it("should render as an unordered list with two items", () => {
    const wrapper = getWrapper();
    const ul = wrapper.find("ul");
    expect(ul.exists()).toBe(true);
    expect(ul.get(0).props.children.length).toEqual(2);
  });

  it("renders a snapshot properly", () => {
    const wrapper = getWrapper();
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
