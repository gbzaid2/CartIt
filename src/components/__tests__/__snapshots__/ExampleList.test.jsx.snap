// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleList renders a snapshot properly 1`] = `
<ul>
  <li>
    Example Item #1
  </li>
  <li>
    Example Item #2
  </li>
</ul>
`;
